### PR TITLE
Simulator

### PR DESCRIPTION
**Pull request by ClarkPoon**
All of the simulator files need to be reviewed (even the parts that are not marked as changed). The previous merge of simulator into master was not reviewed as thoroughly as I would have liked. Also, since this is a massive pull request, I expect all reviewers to help with reviewing this. I would like to get this reviewed soon, so we can get the most up-to-date simulator (and the open wire ) onto the rtos branch.

### CODE REVIEW

File: BSP/Simulator/Src/BSP_UART.c [hunk 1]
```diff
@@ -31,6 +31,12 @@ void BSP_UART_Init(void) {
     setvbuf(stdin, NULL, _IONBF, 0);
     setvbuf(stdout, NULL, _IONBF, 0);
 
+    err = pthread_mutex_init(&rx_mutex, NULL);
```
Comment (manthanand) [inline BSP/Simulator/Src/BSP_UART.c:34]: The CLI/BLE PR also edits this file. If we want to merge that one after simulator we can. Just to let you know though.

Comment (SijWoo) [reply BSP/Simulator/Src/BSP_UART.c:34]: Tank for letting us know. We'll merge after simulator is done

**Review by ClarkPoon — COMMENTED** [review]
I have some minor coding style suggestions. I am also fine merging this if we consider it "good enough" since the simulator is not safety-critical.

File: Apps/Src/Voltage.c [hunk 2]
```diff
@@ -99,15 +99,30 @@ SafetyStatus Voltage_CheckStatus(void){
...
  */
 SafetyStatus *Voltage_GetModulesInDanger(void){
-	static SafetyStatus checks[NUM_BATTERY_MODULES];
-	uint32_t open_wires = Voltage_GetOpenWire();
+	static SafetyStatus checks[TOTAL_PINS];
+	Voltage_GetOpenWire();
+	uint32_t wires;
+	uint32_t openWireIdx = 0;
+	//put all the bits from each minion's system_open_wire variable into one variable
+	for(int k = 0; k < NUM_MINIONS; k++){
```
[the remark this answers is not in the document]
Comment (SijWoo) [reply Apps/Src/Voltage.c:107]: Just for documentation, this function and open wires will be edited in different issue ticket

**Review by SijWoo — DISMISSED** [review]
Some functions have to be edited and those same functions' syntax/formatting will be edited in a different PR but besides that code runs on my computer.

**Review by SijWoo — APPROVED** [review]
Formats and renames variables for better readability. All gud for now!

**Review by SijWoo — APPROVED** [review]
Fixed merge conflict.

**Pull request merged**